### ISSUE TITLE
Support GitHub Actions

### ISSUE THREAD

**Issue by j-f1** [codecov/codecov-node#118]
This would make it much easier to switch CI to run on GitHub. This woked for me:

```hcl
action "Push Coverage" {
  uses = "docker://node:11"
  needs = ["Only Branch Pushes"]
  secrets = ["CODECOV_TOKEN"]
  runs = "bash -c"
  args = ["npx codecov --disable=detect --commit=$GITHUB_SHA --branch=${GITHUB_REF#refs/heads/}"]
}
```

It would be even cooler if you offered an Action that could be run on any project (Node or otherwise)!

```hcl
action "Push Coverage" {
  uses = "codecov/action@master"
  needs = ["Test"]
  secrets = ["CODECOV_TOKEN"]
}
```

**Comment by JasonEtco** [comment]
Awesome to hear! I came to this issue with the same ask. For what it's worth, this can be achieved pretty easily (here's some imperfect pseudocode):

```docker
FROM node
RUN npm i -g codecov
COPY codecov.sh /usr/bin/codecov.sh
ENTRYPOINT ["codecov.sh"]
```

`codecov.sh`:
```shell
#!/bin/sh

set -e

if [ -z "$CODECOV_TOKEN" ]; then
  echo "You must pass a CODECOV_TOKEN secret!"
  exit 1
fi

codecov --disable=detect --commit=$GITHUB_SHA --branch=${GITHUB_REF#refs/heads/}
```

And that's mostly it! This is by no means a perfect solution (you could probably do it all in the Docker container), just figured I'd share a potential start.

**Comment by JasonEtco** [comment]
Sorry for the double post but - here's a little action you can drop into an existing workflow that "just works":

```workflow
action "codecov" {
  uses = "docker://node"
  runs = "npx"
  args = "codecov"
  secrets = ["CODECOV_TOKEN"]
}
```

**Comment by shinnn** [comment]
> Github Actions is in the product pipeline for this quarter

@jerrode May I ask the current progress on it? 

The Codecov FAQ says https://docs.codecov.io/docs/frequently-asked-questions#section-where-is-the-repository-upload-token-found-:

> The upload token is required for all uploads, except originating from public project using Travis-CI, Circle CI, or AppVeyor CI.

It would be nice if "GitHub Actions" is added to those exceptions and we don't need to add `secrets = ["CODECOV_TOKEN"]` to our settings.

**Comment by domoritz** [comment]
@jerrode do you have an update for this? Codecov still doesn't seem to work on github actions without a token (which means it doesn't work in forks for PRs).

**Comment by potiuk** [comment]
Any update on that? We really need it for Apache Airflow - we moved to Github Actions without support for CodeCov but we would love to get it back. Using token is not a good idea for us. Secrets in Apache projects are managed by infrastructure and it's not very convenient to set it up/maintain.

**Comment by eddiemoore** [comment]
There is a Github action for codecov here: https://github.com/codecov/codecov-action For anyone that needs it

**Comment by domoritz** [comment]
Hmm, it doesn't seem to work @eddiemoore without tokens. even though the docs say "The latest release of this Action adds support for tokenless uploads from GitHub Actions!". 

See https://github.com/vega/vega-lite/pull/6768/checks?check_run_id=943398294

<img width="835" alt="Screen Shot 2020-08-03 at 22 00 31" src="https://user-images.githubusercontent.com/589034/89254745-c306b080-d5d4-11ea-8d20-c1dd2f028455.png">

**Comment by thomasrockhu** [comment]
@domoritz, we are currently experiencing an issue detailed [here](https://github.com/codecov/codecov-action/issues/102)

tl;dr, GitHub is not properly setting the `GITHUB_ACTION` environment variable. We will deploy a change soon to use `GITHUB_ACTIONS` instead.

**Comment by domoritz** [comment]
Ahh, thanks @thomasrockhu

**Comment by thomasrockhu** [comment]
We are expecting this to be resolved by 10AM ET as documented by [GitHub](https://github.community/t/github-action-environment-variable-missing-breaking-issue/125913)

**Comment by potiuk** [comment]
Yep. Got the same issue yesterday in my PR above. Waiting impatienttly to get it working :). Thanks for info @thomasrockhu !

**Comment by thomasrockhu** [comment]
@potiuk looks like `codecov-action` should be fixed now. We'll continue to keep this issue open and incorporate GitHub Actions.

**Comment by domoritz** [comment]
Works! Should we close this issue then?

**Comment by potiuk** [comment]
Works as well :+1:

**Comment by thomasrockhu** [comment]
Fixed via https://github.com/codecov/codecov-node/commit/1109eefb8c266bfc52092f93c1a50c82242c9bfd
